### PR TITLE
Patch find sorted items

### PR DESCRIPTION
**Pull request by DOMEYD**
Prevent a bug when the relation named with "_order".

### CODE REVIEW

**Comment by smeeckaert** [comment]
This won't work if the order_property is not defined but if there is *_order field defined.

**Comment by shaoshiva** [comment]
:+1:

**Comment by Meroje** [comment]
:-1: we already have the ability to define the field's name with the `order_field` option https://github.com/novius/novius_renderers/blob/510b4b3396409a30064c686f0bcd753eda70411d/classes/renderer/hasmany.php#L149  
imo the bug is that this option is not used on the jquery selector https://github.com/novius/novius_renderers/blob/510b4b3396409a30064c686f0bcd753eda70411d/static/js/hasmany.js#L11

**Comment by shaoshiva** [comment]
@Meroje This bug is not related to this PR, it was already done this way before.

The `order_field` option should be removed in another PR, it makes no sense to change the field name of the order property.

File: classes/renderer/hasmany.php [hunk 1]
```diff
@@ -287,11 +287,22 @@ protected static function getFieldSet($config, $item)
...
      */
-    protected static function getConfig($item, $data)
+    protected static function getConfig($item, $data, &$renderer_options)
     {
         $class       = get_class($item);
         $config_file = \Config::configFile($class);
         $config      = \Config::load(implode('::', $config_file), true);
+
+        // fallback old
+        if (\Arr::get($renderer_options, 'order', false) && empty(\Arr::get($renderer_options, 'order_property', false))) {
```
Comment (Meroje) [inline classes/renderer/hasmany.php]: This option (`order`) currently does not exist, you should update the docs too (but isn't it redundant with `order_property` ?)

Comment (shaoshiva) [inline classes/renderer/hasmany.php]: Indeed it should be in the doc, but this option already exists.

**Comment by smeeckaert** [comment]
The `order_field` option is only used to store the order value in an other field that the one used by the renderer.

That way you could display a field name `something_order` (since with the current js the field has to be name _order) and store it in something like `something_position`.

If you were to use the `order_field` in the javascript, you would break the compatibility of older instance of the renderer where the `order_field` field doesn't actually exists in the model's configuration and couldn't be found by the javascript. The only way to avoid this issue would be to remove the `*_order` field for every renderer with the `order_field` property set and to create a new one with the `order_field` name, but, then again, you can't really remove fields from existing configuration so it wouldn't be acceptable anyway.

The `order_field` was introduced as a somewhat dirty hack when we couldn't use the before_save method on the fields and we wouldn't to change the javascript. In retrospect it was a bad idea, albeit a fast and working one, that this PR is trying to fix.

File: classes/renderer/hasmany.php [hunk 2]
```diff
@@ -235,7 +235,7 @@ public function before_save($item, $data)
 
             // Trigger the before save of all the fields of the has_many
             $config = static::getConfig($relatedItem, array());
-            $fieldset = static::getFieldSet($config, $relatedItem);
+            $fieldset = static::getFieldSet($config, $relatedItem, $this->renderer_options);
```
Comment (Meroje) [inline classes/renderer/hasmany.php]: this is not needed, the signature is `protected static function getFieldSet($config, $item)`

File: classes/renderer/hasmany.php (hunk 1 above)
Comment (shaoshiva) [inline classes/renderer/hasmany.php]: http://php.net/manual/en/function.empty.php#refsect1-function.empty-parameters

> Prior to PHP 5.5, empty() only supports variables; anything else will result in a parse error. In other words, the following will not work: empty(trim($name)). Instead, use trim($name) == false.

I suggest :

```
if (\Arr::get($renderer_options, 'order', false) && !\Arr::get($renderer_options, 'order_property', false)) {
    //...
}
```

File: classes/renderer/hasmany.php [hunk 3]
```diff
@@ -287,11 +287,22 @@ protected static function getFieldSet($config, $item)
...
     {
         $class       = get_class($item);
         $config_file = \Config::configFile($class);
         $config      = \Config::load(implode('::', $config_file), true);
+
+        // fallback old
+        if (\Arr::get($renderer_options, 'order', false) && !\Arr::get($renderer_options, 'order_property', false)) {
+            foreach (\Arr::get($config, 'fieldset_fields') as $k => $field) {
+                if (preg_match('/(.*)_order(.*)/', $k)) {
+                    \Arr::set($renderer_options, 'order_property', true);
```
Comment (Meroje) [inline classes/renderer/hasmany.php]: Shouldn't that be set to the name of the field `$k` ?

**Comment by Meroje** [comment]
:+1: lgtm

**Pull request merged**